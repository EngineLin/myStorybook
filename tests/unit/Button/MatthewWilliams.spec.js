import { shallowMount } from '@vue/test-utils'

import MatthewWilliamsComponent from '@/stories/Button/MatthewWilliams.vue';

const initData = () => ({
    label: 'new message',
    value: 'value with string'
})

const defaultData = {
    label: 'View my work',
    value: true
}

it('Match snapshot', () => {
    const data = initData()
    const wrapper = shallowMount(MatthewWilliamsComponent, {
        propsData: { data }
    })
    expect(wrapper.html()).toMatchSnapshot()
})

it('Renders default props.label if no passed', () => {
    const wrapper = shallowMount(MatthewWilliamsComponent)
    expect(wrapper.text()).toEqual(defaultData.label)
})

it('Renders props.label when passed', () => {
    const data = initData()
    const wrapper = shallowMount(MatthewWilliamsComponent, {
        propsData: { data }
    })
    expect(wrapper.text()).toEqual(data.label)
})

it('Event click invoke with props.value if no passed', async () => {
    const wrapper = shallowMount(MatthewWilliamsComponent)
    await wrapper.trigger('click')
    expect(wrapper.emitted().click[0][0]).toEqual(defaultData.value)
})

it('Event click invoke with props.value', async () => {
    const data = initData()
    const wrapper = shallowMount(MatthewWilliamsComponent, {
        propsData: { data },
    })
    await wrapper.trigger('click')
    expect(wrapper.emitted().click[0][0]).toEqual(data.value)
})
