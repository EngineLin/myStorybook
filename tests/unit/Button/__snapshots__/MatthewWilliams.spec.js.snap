// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
"<button class=\\"button px-5 py-3
               flex items-center
               border-2 border-white
               hover:bg-teal-400 hover:border-teal-400 focus:outline-none\\">
  <p class=\\"text-xl text-white tracking-wide\\">
    new message</p> <span class=\\"icon-container
               h-full ml-4 mr-2
               flex justify-center items-center\\"><i class=\\"icon ion-md-arrow-round-forward
              text-white text-lg\\"></i></span>
</button>"
`;
